core/os/os.h

```diff
@@ -159,7 +159,7 @@ class OS {
 	virtual Error kill(const ProcessID &p_pid) = 0;
 	virtual int get_process_id() const;
 	virtual bool is_process_running(const ProcessID &p_pid) const = 0;
-	virtual void vibrate_handheld(int p_duration_ms = 500);
+	virtual void vibrate_handheld(int p_duration_ms = 500) = 0;
```

Comment:
`= 0` makes it pure virtual and requires an implementation in all platforms.
```suggestion
	virtual void vibrate_handheld(int p_duration_ms = 500) {}
```